api_tests/preprints/views/test_preprint_versions.py

```diff
@@ -154,3 +155,377 @@ def test_permission_error(self):
+        assert res.status_code == 200
+        data = res.json['data']
+        assert len(data) == 7
+        assert len(set([item['id'] for item in data])) == 7
+
+    def test_preprints_version_permissions_for_admin(self):
+        res = self.app.get(self.version_list_url, auth=self.admin.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 6
+
+        unpublished_preprint_version = PreprintFactory.create_version(
+            create_from=self.latest_preprint,
+            creator=self.user,
+            final_machine_state='initial',
+            is_published=False,
+            set_doi=False
+        )
+        self.latest_preprint = unpublished_preprint_version
+        unpublished_preprint_version.add_contributor(self.admin, permissions=permissions.ADMIN)
+        res = self.app.get(self.version_list_url, auth=self.admin.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 7
+
+        unpublished_preprint_version.run_submit(user=self.admin)
+        res = self.app.get(self.version_list_url, auth=self.admin.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 7
+
+        unpublished_preprint_version.run_accept(user=self.admin, comment='Text')
+
+        res = self.app.get(self.version_list_url, auth=self.admin.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 7
+
+        unpublished_version = PreprintFactory.create_version(
+            create_from=self.latest_preprint,
+            creator=self.user,
+            final_machine_state='initial',
+            is_published=False,
+            set_doi=False
+        )
+        unpublished_version.add_contributor(self.admin, permissions=permissions.ADMIN)
+
+        unpublished_version.run_submit(user=self.admin)
+
+        res = self.app.get(self.version_list_url, auth=self.admin.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 8
+
+        unpublished_version.run_reject(user=self.admin, comment='Test')
+        res = self.app.get(self.version_list_url, auth=self.admin.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 8
+
+    def test_preprints_version_permissions_for_write_user(self):
+        new_user = AuthUserFactory()
+        res = self.app.get(self.version_list_url, auth=new_user.auth)
+        assert res.status_code == 200
+        assert len(res.json['data']) == 6
+
+        unpublished_preprint_version = PreprintFactory.create_version(
+            create_from=self.latest_preprint,
+            creator=self.user,
+            final_machine_state='initial',
+            is_published=False,
+            set_doi=False
+        )
+        self.latest_preprint = unpublished_preprint_version
+        unpublished_preprint_version.add_contributor(self.admin, permissions=permissions.ADMIN)
+        unpublished_preprint_version.add_contributor(new_user, permissions=permissions.WRITE)
```

Comment:
Ditto, let's set up the contributors correctly as I mentioned earlier so we don't have to add new contributor